File: 1260.c
#include <cstdio>
#include <cstdlib>
#include <set>
#include <queue>
using namespace std;

void bfs(int **graph, int start_v, int n);
void dfs(int **graph, int v, int n);

queue<int> bfsqueue;
set<int> visited;

int main() {
	int n, m, v;
	int **graph;
	int v1, v2;

	scanf("%d%d%d", &n, &m, &v);

	graph = (int **) malloc(n * sizeof(int *));
	for (int i = 0; i < n; ++i) {
		graph[i] = (int *) calloc(n ,sizeof(int));
	}

	for (int i = 0; i < m; ++i) {
		scanf("%d%d", &v1, &v2);
		graph[v1 - 1][v2 - 1] = 1;
		graph[v2 - 1][v1 - 1] = 1;
	}

	dfs(graph, v - 1, n);
	printf("\n");
	bfs(graph, v - 1, n);

	return 0;
}

void dfs(int **graph, int v, int n) {
	visited.insert(v);
	printf("%d ", v + 1);
	for (int i = 0; i < n; ++i) {
		if (graph[v][i]) {
			if (visited.find(i) == visited.end()) {
				dfs(graph, i, n);
			}
		}
	}
}

void bfs(int **graph, int start_v, int n) {
	int current_v;
	bfsqueue.push(start_v);
	visited.clear();
	visited.insert(start_v);
	
	while (!bfsqueue.empty()) {
		current_v = bfsqueue.front();
		bfsqueue.pop();
		printf("%d ", current_v + 1);
		for (int i = 0; i < n; ++i) {
			if (graph[current_v][i]) {
				if (visited.find(i) == visited.end()) {
					visited.insert(i);
					bfsqueue.push(i);
				}
			}
		}

	}

	printf("\n");
}

File: 1260.py
# DFS와 BFS
# https://www.acmicpc.net/problem/1260

import sys


def dfs(graph, visited, current_node):
    visited[current_node] = True
    print(current_node, end=" ")

    for next_node in graph[current_node]:
        if not visited[next_node]:
            dfs(graph, visited, next_node)


def bfs(graph, start_node):
    bfs_queue = []
    visited = [False] * (len(graph) + 1)
    bfs_queue.append(start_node)
    visited[start_node] = True

    while len(bfs_queue) > 0:
        current_node = bfs_queue.pop(0)

        print(current_node, end=" ")

        for next_node in graph[current_node]:
            if not visited[next_node]:
                visited[next_node] = True
                bfs_queue.append(next_node)


n, m, v = map(int, sys.stdin.readline().split())
graph = [[] for _ in range(n + 1)]
visited = [False] * (n + 1)

for _ in range(m):
    a, b = map(int, sys.stdin.readline().split())
    graph[a].append(b)
    graph[b].append(a)

for subgraph in graph:
    subgraph.sort()

dfs(graph, visited, current_node=v)
print()
bfs(graph, start_node=v)
print()